codeface_extraction/codeface_extraction.py

```diff
@@ -50,34 +50,39 @@ def run_extraction(conf, resdir, extract_impl, extract_on_range_level):
     dbm = DBManager(conf)
 
     # get all types of extractions, both project-level and range-level
-    __extractions_project, __extractions_range = extractions.get_extractions(dbm, conf, resdir, csv_writer, extract_impl, extract_on_range_level)
+    __extractions_project, __extractions_range = extractions.get_extractions(dbm, conf, resdir, csv_writer,
+                                                                             extract_commit_messages, extract_impl,
+                                                                             extract_on_range_level)
 
     # run project-level extractions
     for extraction in __extractions_project:
         extraction.run()
 
-    # check if list of revisions in database is the same as in the config file
-    revs = conf["revisions"]
-    list_of_revisions = extractions.RevisionExtraction(dbm, conf, resdir, csv_writer).get_list()
-    if revs:
-        if set(revs) != set(list_of_revisions):
-            log.error("List of revisions in configuration file do not match the list stored in the DB! Stopping now.")
-            sys.exit(1)
+    # run range-level extractions (only if explicitely enabled)
+    if extract_on_range_level:
+
+        # check if list of revisions in database is the same as in the config file
```

Comment:
Do we still need this check (i.e., Lines 64–75)? I added it in a time when we have not introduced the selection processes yet and built networks on release ranges defined/extracted by Codeface – it kept me from overwriting old extractions for casestudies where I changed the configuration file, but forgot to run the commit analysis in Codeface. As we do not construct networks on Codeface ranges most of the time, we may want to remove this check.

Reply:
Nonetheless, I would like to keep the check as it does not hurt and if someone eventually wants to use the Codeface ranges (for whatever reason), it will be a useful check.